src/loaders/FileLoader.js

```diff
@@ -196,6 +196,15 @@ class FileLoader extends Loader {
 				// Abort errors and other errors are handled the same
 
 				const callbacks = loading[ url ];
+
+				if ( callbacks === undefined ) {
+
+					// When onLoad was called and url was deleted in `loading`
+					this.manager.itemError( url );
```

Comment:
I think we need to fire `this.manager.itemEnd( url );` here as it does at the end of the "catch" block, too

Reply:
In the mentioned case `itemEnd()` is then executed twice. At this place and in the previous `then()` block.

Reply:
I don't think that's the case. If an error is thrown in the promise callbacks and the catch callback is run then the above itemEnd will not be hit. And either way the catch function already calls the itemEnd function at the bottom of it so it should be called in this new branch too for consistency.

Another option would be to call itemEnd in a ".finally" callback, instead, so it always gets run

Reply:
> Another option would be to call itemEnd in a ".finally" callback, instead, so it always gets run

That sounds like the cleanest solution to me.

Reply:
DONE

Reply:
Great! I think this looks right to me, now!